src/Microsoft.VisualStudio.Composition/Resolver.cs

```diff
@@ -36,5 +41,108 @@ public Resolver(IAssemblyLoader assemblyLoader)
         }
 
         internal IAssemblyLoader AssemblyLoader { get; }
+
+        /// <summary>
+        /// Gets identity and version metadata for an assembly with the given <see cref="AssemblyName"/>,
+        /// if that assembly has been loaded with this <see cref="Resolver"/>.
+        /// </summary>
+        /// <param name="assemblyName">The name of the assembly to look up.</param>
+        /// <param name="assemblyId">Receives the metadata from the assembly, if it has been loaded by this <see cref="Resolver"/>; otherwise <c>null</c>.</param>
+        /// <returns><c>true</c> if the metadata was found; <c>false</c> otherwise.</returns>
+        internal bool TryGetAssemblyId(AssemblyName assemblyName, out StrongAssemblyIdentity assemblyId)
+        {
+            lock (this.loadedAssemblyStrongIdentities)
+            {
+                return this.loadedAssemblyStrongIdentities.TryGetValue(assemblyName, out assemblyId);
+            }
+        }
+
+        /// <summary>
+        /// Determines the strong identity of an assembly and stores it.
+        /// </summary>
+        /// <param name="assembly">The loaded assembly.</param>
+        /// <param name="assemblyName">An optional <see cref="AssemblyName"/> that may be important for dynamic assemblies to find their CodeBase.</param>
+        /// <returns>The identity determined for this assembly.</returns>
+        internal StrongAssemblyIdentity GetStrongAssemblyIdentity(Assembly assembly, AssemblyName assemblyName) => this.NotifyAssemblyLoaded(assembly, assemblyName);
+
+        /// <summary>
+        /// Determines the strong identity of an assembly and stores it.
+        /// </summary>
+        /// <param name="assembly">The loaded assembly.</param>
+        /// <param name="assemblyName">An optional <see cref="AssemblyName"/> that may be important for dynamic assemblies to find their CodeBase.</param>
+        /// <returns>The identity determined for this assembly.</returns>
+        private StrongAssemblyIdentity NotifyAssemblyLoaded(Assembly assembly, AssemblyName assemblyName)
+        {
+            Requires.NotNull(assembly, nameof(assembly));
+
+            if (assemblyName == null)
+            {
+                assemblyName = assembly.GetName();
+            }
+
+            lock (this.loadedAssemblyStrongIdentities)
+            {
+                if (this.loadedAssemblyStrongIdentities.TryGetValue(assemblyName, out var result))
+                {
+                    return result;
+                }
+            }
```

Comment:
just call TryGetAssemblyId here?

Reply:
I'm fixing this in a follow up merge that is validating now